Microsoft.Toolkit.Uwp.UI.Controls.Primitives/ConstrainedBox/ConstrainedBox.Properties.cs

```diff
@@ -93,11 +93,71 @@ public AspectRatio AspectRatio
         public static readonly DependencyProperty AspectRatioProperty =
             DependencyProperty.Register(nameof(AspectRatio), typeof(AspectRatio), typeof(ConstrainedBox), new PropertyMetadata(null, ConstraintPropertyChanged));
 
+        private bool _propertyUpdating;
+
         private static void ConstraintPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
         {
-            if (d is ConstrainedBox panel)
+            if (d is ConstrainedBox panel && !panel._propertyUpdating)
             {
+                panel._propertyUpdating = true;
+
+                panel.CoerceValues();
+
                 panel.InvalidateMeasure();
+
+                panel._propertyUpdating = false;
+            }
+        }
+
+        private void CoerceValues()
+        {
+            // Check if scale properties are in range
+            if (!double.IsNaN(ScaleX))
+            {
+                if (ScaleX < 0)
+                {
+                    ScaleX = 0;
+                }
+                else if (ScaleX > 1.0)
+                {
+                    ScaleX = 1.0;
+                }
+            }
+            else
+            {
+                ScaleX = 1.0;
+            }
+
+            if (!double.IsNaN(ScaleY))
+            {
+                if (ScaleY < 0)
+                {
+                    ScaleY = 0;
+                }
+                else if (ScaleY > 1.0)
+                {
+                    ScaleY = 1.0;
+                }
+            }
+            else
+            {
+                ScaleY = 1.0;
+            }
+
+            // Clear invalid values less than 0 for other properties
+            if (ReadLocalValue(MultipleXProperty) is int value && value <= 0)
+            {
+                ClearValue(MultipleXProperty);
```

Comment:
In our tests we test against DP.UnsetValue, wouldn't it be better to have the properties reflect the value we would want in that case? Also are there any tests for the case that those properties are set to zero?

Reply:
Not sure if I follow, the `ClearValue` here will set it back to `UnsetValue`? Are you just saying that I should have a test that checks for setting the invalid value and that this happens?

Reply:
What would be the unset value be for that property? And yes a test that would check for an invalid value would be good but also not necessary.

Reply:
The default is already Unset, I check for that here:

https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/2d740769c1a39ced967aa488b16602510d7fdefd/Microsoft.Toolkit.Uwp.UI.Controls.Primitives/ConstrainedBox/ConstrainedBox.cs#L118-L135

So we only apply it if a developer has applied a value to the property. All this is doing in the coercion check is if they're providing a negative number instead (or 0) which wouldn't make sense to reset it back to unset by calling Clear.

Reply:
I see, guess it's fine then. LGTM